packages/utils/src/parser/token.spec.ts

```diff
@@ -336,13 +336,9 @@ describe("TokenAmountV2 with 18 decimals", () => {
     ).toBe(FromStringToTokenError.InvalidFormat);
   });
 
-  it("does not support scientific notation", () => {
-    const callToNumber = () =>
-      TokenAmountV2.fromNumber({ token: token, amount: 1e-9 });
-    expect(callToNumber).toThrow(
-      expect.objectContaining({
-        message: "Invalid number 1e-9",
-      }),
+  it("truncates scientific notation to 8 decimals", () => {
```

Comment:
Actually, I think the name of this test does not match what the test does, both now and before.

Supporting scientific notation would have to be done in `.fromString()`. When using `.fromNumber`, there is no such thing as "notation".
`1e-9` and `0.000000001` are 2 different ways of writing *the same* number in JS/TS but from the perspective of the `.fromNumber()` method there is no difference.

I think we should have 2 separate tests:
1. `fromNumber({amount: 0.00000001})` works (and does not get converted to "1e-8" in an intermediate step.
2. `fromString("1e-8")` does not work.

Only for the second one does it make sense to have the description be about "scientific notation".

Reply:
That makes sense.

I changed the tests. Let me know what you think.